qiskit_finance/data_providers/_base_data_provider.py

```diff
@@ -42,42 +57,60 @@ class BaseDataProvider(ABC):
 
     To use the subclasses, please see
     https://qiskit-community.github.io/qiskit-finance/tutorials/11_time_series.html
-
     """
 
     @abstractmethod
     def __init__(self) -> None:
-        self._data: Optional[List] = None
-        self._n = 0  # pylint: disable=invalid-name
-        self.period_return_mean: Optional[np.ndarray] = None
-        self.cov: Optional[np.ndarray] = None
-        self.period_return_cov: Optional[np.ndarray] = None
-        self.rho: Optional[np.ndarray] = None
-        self.mean: Optional[np.ndarray] = None
+        self._data: list | None = None
+        self._n: int = 0  # pylint: disable=invalid-name
+        self.period_return_mean: np.ndarray | None = None
+        self.cov: np.ndarray | None = None
+        self.period_return_cov: np.ndarray | None = None
+        self.rho: np.ndarray | None = None
+        self.mean: np.ndarray | None = None
 
     @abstractmethod
     def run(self) -> None:
-        """Loads data."""
+        """
+        Abstract method to load data.
+
+        Method responsible for loading data. Subclasses of `BaseDataProvider`
+        must implement this method to load data from a specific data source.
+
+        Raises:
+            NotImplementedError: This method must be implemented in a subclass.
```

Comment:
The code here does not raise this exception so it seems wrong in the docs here. If the subclass does not implement this method then Python will refuse to instantiate the subclass since it will still be abstract (same deal if you try and instantiate this base class)